### PR TITLE
Optimize projectiles 01

### PR DESCRIPTION
**Pull request by Ejsstiil**
https://github.com/FAForever/fa/pull/3724

Optimize following projectiles. Unused code, added LODCutoff where applicable.
TIFMortar01
TIFNapalmCarpetBomb01
TIFNapalmCarpetBomb02
TIFSmallYieldNuclearBomb01
TIMAntiMissile01
TIMAntiTorpedo01
TIMAntiTorpedo02
TIMMissileIntercerptor01
TIMSonarBuoyCreator01
TorpedoShipAeon01
TorpedoShipTerran01
TorpedoShipTerran02
TorpedoShipTerran03
TorpedoSubTerran01

### CODE REVIEW

File: projectiles/TIFMortar01/TIFMortar01_script.lua [hunk 1]
```diff
@@ -1,9 +1,4 @@
---
 -- Terran Mortar
---
-local TArtilleryProjectile = import('/lua/terranprojectiles.lua').TArtilleryProjectile
-TIFMortar01 = Class(TArtilleryProjectile) {
-}
 
+TIFMortar01 = Class(import('/lua/terranprojectiles.lua').TArtilleryProjectile) {}
```
Comment (speed2CZ) [inline projectiles/TIFMortar01/TIFMortar01_script.lua:3]: putting the import into the Class call makes it less readable

File: projectiles/TIFNapalmCarpetBomb01/TIFNapalmCarpetBomb01_script.lua [hunk 2]
```diff
@@ -1,9 +1,7 @@
...
 -- Terran Napalm Carpet Bomb
---
-
-local TNapalmCarpetBombProjectile = import('/lua/terranprojectiles.lua').TNapalmCarpetBombProjectile
-
-TIFNapalmCarpetBomb01 = Class(TNapalmCarpetBombProjectile) { }
 
+--unused???
+--[[
+TIFNapalmCarpetBomb01 = Class(import('/lua/terranprojectiles.lua').TNapalmCarpetBombProjectile) { }
```
Comment (speed2CZ) [inline projectiles/TIFNapalmCarpetBomb01/TIFNapalmCarpetBomb01_script.lua:5]: they can be used by mods, don't comment them out

**Review by speed2CZ — CHANGES_REQUESTED** [review]

File: projectiles/TIFNapalmCarpetBomb02/TIFNapalmCarpetBomb02_script.lua [hunk 3]
```diff
@@ -8,10 +8,6 @@
 --  Copyright © 2007 Gas Powered Games, Inc.  All rights reserved.
 -------------------------------------------------------------------------------
 
-local TNapalmHvyCarpetBombProjectile = import('/lua/terranprojectiles.lua').TNapalmHvyCarpetBombProjectile
```
Comment (Garanas) [inline projectiles/TIFNapalmCarpetBomb02/TIFNapalmCarpetBomb02_script.lua:11]: I'll update the issue / pull request.

All locals that are removed should be placed beneath the class. That prevents them from being part of a closure (yah) while maintaining mod support.

Mod files are 'appended' to the bottom of each file. Therefore they can use the original imports.

File: projectiles/TIFNapalmCarpetBomb01/TIFNapalmCarpetBomb01_script.lua (hunk 2 above)
Comment (Garanas) [reply projectiles/TIFNapalmCarpetBomb01/TIFNapalmCarpetBomb01_script.lua:5]: I agree, the base game doesn't use all projectiles. But a mod may use it.

File: projectiles/TIFNapalmCarpetBomb02/TIFNapalmCarpetBomb02_script.lua (hunk 3 above)
Comment (Ejsstiil) [reply projectiles/TIFNapalmCarpetBomb02/TIFNapalmCarpetBomb02_script.lua:11]: Why then remove them?

Comment (Garanas) [reply projectiles/TIFNapalmCarpetBomb02/TIFNapalmCarpetBomb02_script.lua:11]: Good point 😄

**Comment by Garanas** [comment]
Made a decision, see also:

image

At no point should we remove top-file locals, we can move them to the bottom of the file however. At any point all the top-file definitions (outer scope) should remain available, as mods may depend on their existence.

**Review by Garanas — APPROVED** [review]
Looking good - fixed some of the comment / annotation style.

**Pull request merged**